### PR TITLE
Implementing Mobile Profile Pages

### PR DESCRIPTION
**Pull request by vindema**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/86)
Implemented the Profile and Settings screens for the mobile app, alongside reusable UI components to standardize the design.

# How Has This Been Tested?
- Manual UI Testing: Verified that the Profile and Settings screens render correctly in both Light and Dark modes.
- Navigation Testing: Clicked through the list items on the Profile screen to ensure proper routing to `/payment`, `/family`, and `/settings`. Verified the back button on the Settings screen works correctly.

## Screenshots
Please provide screenshots of manual testing (Scalar, frontend pages, etc.)

# Checklist

## General
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [x] I have written unit tests for my code
- [x] I have written integration tests where appropriate
- [x] New and existing unit tests pass locally with my changes
- [x] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [x] I have tested error handling and edge cases
- [x] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [x] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [x] Mobile responsive design verified (if applicable)
- [x] I have run make generate-api to keep the API hooks up to date
- [x] **For mobile features:** Tested on both iOS and Android devices
- [x] **For mobile features:** Screenshots from both iOS and Android included below
- [x] No console errors or warnings
- [x] Loading states and error states handled

### iOS
Screenshot 2026-03-13 at 4 38 05 PM
Screenshot 2026-03-13 at 4 38 18 PM
Screenshot 2026-03-13 at 4 38 29 PM
Screenshot 2026-03-13 at 4 38 40 PM
Screenshot 2026-03-13 at 11 55 18 PM
Screenshot 2026-03-13 at 11 55 31 PM
Screenshot 2026-03-13 at 11 57 38 PM

### Android
Screenshot 2026-03-15 at 2 11 12 AM
Screenshot 2026-03-15 at 2 11 27 AM
Screenshot 2026-03-15 at 2 11 58 AM
Screenshot 2026-03-15 at 2 20 20 AM
Screenshot 2026-03-15 at 2 20 51 AM
Screenshot 2026-03-15 at 2 21 05 AM
Screenshot 2026-03-15 at 2 21 25 AM

### CODE REVIEW

File: frontend/apps/mobile/app/(tabs)/family/manage.tsx [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+
+const GUARDIAN_ID = '88888888-8888-8888-8888-888888888888';
+
+const INTEREST_OPTIONS = [
+  'science', 'math', 'music', 'art', 'sports', 'technology', 'language', 'other',
+];
+
+const capitalize = (s: string) => s.charAt(0).toUpperCase() + s.slice(1);
+
+const TAG_COLORS = [
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(tabs)/family/manage.tsx]: I think these colors should be abstracted out to a constants file

File: frontend/apps/mobile/app/(tabs)/family/manage.tsx [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+                <ThemedText className="text-[#EF4444] font-nunito-semibold">Delete</ThemedText>
+              </TouchableOpacity>
+            ) : (
+              <View className="w-10" />
+            )}
+          </View>
+          <ThemedText className="text-[22px] font-nunito-semibold mb-5">
+            {isEditing ? 'Edit Child Profile' : 'Create Child Profile'}
+          </ThemedText>
+          <TextInput
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(tabs)/family/manage.tsx]: Can you look into making these in a form component
